src/pathbar.cpp

```diff
@@ -312,6 +312,9 @@ void PathBar::closeEditor() {
     if(tempPathEdit_ == nullptr) {
         return;
     }
+    // Avoid recursive calling of this slot from nested signals
+    // see https://github.com/lxqt/libfm-qt/issues/225
+    tempPathEdit_->disconnect();
     // If a menu has popped up synchronously (with QMenu::exec), the path buttons may be drawn
     // but the path-edit may not disappear until the menu is closed. So, we hide it here.
```

Comment:
Nice catch! Would you please move `tempPathEdit_->disconnect();` here, below the hiding comment but above the hiding line and add this to the hiding comment instead:
```c++
    // But since hiding the path-edit makes it lose focus and emit editingFinished(),
    // we should disconnect from it to avoid recursive calling of the current function.
```

Reply:
done